java/flight/flight-sql-jdbc-core/src/test/java/org/apache/arrow/driver/jdbc/ConnectionMutualTlsTest.java

```diff
@@ -0,0 +1,436 @@
+  public void testGetEncryptedClientWithBadMTlsKeyPath() throws Exception {
+    final UsernamePasswordCredentials credentials = new UsernamePasswordCredentials(
+            userTest, passTest);
+
+    try (ArrowFlightSqlClientHandler ignored =
+                 new ArrowFlightSqlClientHandler.Builder()
+                         .withHost(FLIGHT_SERVER_TEST_RULE.getHost())
+                         .withPort(FLIGHT_SERVER_TEST_RULE.getPort())
+                         .withUsername(credentials.getUserName())
+                         .withPassword(credentials.getPassword())
+                         .withTlsRootCertificates(tlsRootCertsPath)
+                         .withClientCertificate(clientMTlsCertPath)
+                         .withClientKey(badClientMTlsKeyPath)
+                         .withBufferAllocator(allocator)
+                         .withEncryption(true)
+                         .build()) {
+      Assert.fail();
+    }
+  }
+
+  /**
+   * Try to instantiate an encrypted FlightClient without credentials.
+   *
+   * @throws Exception on error.
+   */
+  @Test
+  public void testGetNonAuthenticatedEncryptedClientNoAuth() throws Exception {
+    try (ArrowFlightSqlClientHandler client =
+             new ArrowFlightSqlClientHandler.Builder()
+                 .withHost(FLIGHT_SERVER_TEST_RULE.getHost())
+                 .withTlsRootCertificates(tlsRootCertsPath)
+                 .withClientCertificate(clientMTlsCertPath)
+                 .withClientKey(clientMTlsKeyPath)
+                 .withBufferAllocator(allocator)
+                 .withEncryption(true)
+                 .build()) {
+      assertNotNull(client);
+    }
+  }
+
+  /**
+   * Check if an encrypted connection can be established successfully when the
+   * provided valid credentials and a valid TLS Root Certs path.
+   *
+   * @throws Exception on error.
+   */
+  @Test
+  public void testGetEncryptedConnectionWithValidCredentialsAndTlsRootsPath() throws Exception {
+    final Properties properties = new Properties();
+
+    properties.put(ArrowFlightConnectionProperty.HOST.camelName(), "localhost");
+    properties.put(ArrowFlightConnectionProperty.PORT.camelName(),
+        FLIGHT_SERVER_TEST_RULE.getPort());
+    properties.put(ArrowFlightConnectionProperty.USER.camelName(),
+        userTest);
+    properties.put(ArrowFlightConnectionProperty.PASSWORD.camelName(),
+        passTest);
+    properties.put(ArrowFlightConnectionProperty.TLS_ROOT_CERTS.camelName(),
+        tlsRootCertsPath);
+    properties.put(ArrowFlightConnectionProperty.CLIENT_CERTIFICATE.camelName(),
+        clientMTlsCertPath);
+    properties.put(ArrowFlightConnectionProperty.CLIENT_KEY.camelName(),
+        clientMTlsKeyPath);
+
+    final ArrowFlightJdbcDataSource dataSource =
+        ArrowFlightJdbcDataSource.createNewDataSource(properties);
+    try (final Connection connection = dataSource.getConnection()) {
+      assert connection.isValid(300);
```

Comment:
Out of curiosity, why is there a 300 second timeout for ArrowFlightJdbcDataSource connections, but 0 second timeouts for `ArrowFlightJdbcDriver`/`DriverManager` connections?

Reply:
Very good question :) - I copied the tests from the `ConnectionTlsTest.java` code - while modifying the tests for mTLS of course.  I'm not sure of the rationale for the timeout, but I'll research it.